tests/unit/src/quaternion.cpp

```diff
@@ -292,12 +314,94 @@ tst::set set("quaternion", [](tst::suite& suite){
         // TODO: test to_matrix4()
     });
 
-    suite.add_disabled("slerp_quaternion_t", []{
-		// auto slow_slerp = [](r4::quaternion<float> a, r4::quaternion<float> b, float t){
-		// 	// 
-		// };
-
-        // TODO: test slerp(quaternion, t)
+    suite.add<std::tuple<
+		r4::quaternion<float>,
+		r4::quaternion<float>,
+		float
+	>>(
+		"slerp",
+		{
+			{
+				r4::quaternion<float>().set_rotation(1, 0, 0, utki::pi<float>() / 2),
+				r4::quaternion<float>().set_rotation(0, 1, 0, utki::pi<float>() / 2),
+				0.001f
+			},
+			{
+				r4::quaternion<float>().set_rotation(1, 0, 0, utki::pi<float>() / 2),
+				r4::quaternion<float>().set_rotation(0, 1, 0, utki::pi<float>() / 2),
+				0.5f
+			},
+			{
+				r4::quaternion<float>().set_rotation(1, 0, 0, utki::pi<float>() / 2),
+				r4::quaternion<float>().set_rotation(0, 1, 0, utki::pi<float>() / 2),
+				0.999f
+			},
+			// test two very close quaternions
+			{
+				r4::quaternion<float>().set_rotation(1, 0, 0, utki::pi<float>() / 2),
+				r4::quaternion<float>().set_rotation(1.01, 0, 0, utki::pi<float>() / 2),
+				0.001f
+			},
+			{
+				r4::quaternion<float>().set_rotation(1, 0, 0, utki::pi<float>() / 2),
+				r4::quaternion<float>().set_rotation(1.01, 0, 0, utki::pi<float>() / 2),
+				0.5f
+			},
+			{
+				r4::quaternion<float>().set_rotation(1, 0, 0, utki::pi<float>() / 2),
+				r4::quaternion<float>().set_rotation(1.01, 0, 0, utki::pi<float>() / 2),
+				0.999f
+			}
+		},
+		[](const auto& p){
+			float eps = 0.01f;
+
+			auto slow_slerp = [eps](r4::quaternion<float> a, r4::quaternion<float> b, float t){
+				tst::check_le(t, decltype(t)(1), SL);
+				tst::check_ge(t, decltype(t)(0), SL);
+
+				// Let slerp(0) = A and slerp(1) = B, then for slerp(t) calculation:
+				// c = (a^-1) % b
+				// angle = acos(c.s) * t
+				// slerp(t) = a % quaternion(c.v * sin(angle) / c.v.norm(), cos(angle))
+
+				auto c = a.inv() * b;
+
+				auto n2 = c.v.norm_pow2();
+				// to avoid division by 0 for small 'n2' we use approximation
+				// for sine and cosine functions:
+				// sin(x) = x, cos(x) = 1 - x^2
+				if(n2 < eps){
```

Comment:
0.01 is compared with sin(a)^2 so sin(a) is compared with 0.1, sin(0.1) ~ 0.998, so error is about 0.002, that could be large error for some applications, so may be lower value of epsilon would be better.

Reply:
0.001?

Reply:
changed to 0.001